### PR TITLE
Support int8 for operator Split

### PR DESCRIPTION
**Pull request by xadupre**
**Description**:
Split did not support int8 in onnxruntime as specified in ONNX specifications.

**Motivation and Context**
- Model yamnet for tflite is using this operator.

### CODE REVIEW

File: onnxruntime/core/providers/cpu/tensor/split.cc [hunk 1]
```diff
@@ -16,7 +16,7 @@ namespace onnxruntime {
 namespace op_kernel_type_control {
 ORT_SPECIFY_OP_KERNEL_ARG_DEFAULT_TYPES_ALL_OPSETS(
     kCpuExecutionProvider, kOnnxDomain, Split, Input, 0,
-    float, int32_t, int64_t, uint8_t, std::string);
+    float, int8_t, int32_t, int64_t, uint8_t, std::string);
```
Comment (skottmckay) [inline onnxruntime/core/providers/cpu/tensor/split.cc:19]: @edgchen1 What's the best way to update whilst preserving hashes when using BuildKernelDefConstraintsFromTypeList?

Comment (edgchen1) [reply onnxruntime/core/providers/cpu/tensor/split.cc:19]: The current way looks good to me.

**Review by skottmckay — DISMISSED** [review]
:shipit:

File: onnxruntime/core/providers/cpu/tensor/split.cc (hunk 1 above)
Comment (xadupre) [reply onnxruntime/core/providers/cpu/tensor/split.cc:19]: docs/OperatorKernels.md needs to be updated as well. That would probably be worth adding it in kernel_def_hash_test.cc.

**Comment by xadupre** [comment]
The Android build was interrupted because too long. I don't think that's related to my changes. Any suggestion to make it pass?

**Comment by edgchen1** [comment]
> The Android build was interrupted because too long. I don't think that's related to my changes. Any suggestion to make it pass?

For now, you can retry it. I have a PR for improving it: https://github.com/microsoft/onnxruntime/pull/8644

**Review by edgchen1 — APPROVED** [review]

**Pull request merged**